TBS Add generic timeout message for Progressive Enrollment (#3358)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/reference/registration-hook/index.md

For information on the API for registering external service endpoints with Okta, see [Inline Hooks Management API](/docs/reference/api/inline-hooks/).

For steps to enable this Inline Hook, see [Enabling a Registration Inline Hook](/docs/guides/registration-inline-hook/nodejs/main/#enable-the-registration-inline-hook).

For an example implementation of this Inline Hook, see [Registration Inline Hook](/docs/guides/registration-inline-hook/nodejs/main/).

## About

The Okta Registration Inline Hook allows you to integrate your own custom code into Okta's [Profile Enrollment](https://help.okta.com/okta_help.htm?type=oie&id=ext-create-profile-enrollment) flow. The hook is triggered after Okta receives the registration or profile update request. Your custom code can:

- Allow or deny the registration attempt, based on your own validation of the information the user has submitted

> - Set or override the values that will be populated in attributes of the user's Okta profile

Reviewer: ```suggestion
- Set or override the values that are populated in attributes of the user's Okta profile
```

Using the `com.okta.action.update` [command](#supported-commands) in your response, you can change the action that Okta will take.

> <!-- Need to clarify if we need to include this in the docs. right now, it has no content. in the context of the registration inline hook guide, it allows customers to include the object in the console.log().

Author: Okay, I'll leave it commented out for now. Thanks!

> <!-- Need to clarify if we need to include this in the docs. right now, it has no content. in the context of the registration inline hook guide, it allows customers to include the object in the console.log().

Reviewer: ```suggestion
<!-- Need to clarify if we need to include this in the docs. right now, it has no content. in the context of the registration inline hook guide, it allows customers to include the object in the console.log().-->
```

Author: See comment on line 98.

### data.context.user

> <ApiLifecycle access="ie" /><br> -->

Author: Hi @susanharper-okta I put the closing comment "tag" here. Without it, we have an orphan title with no content :)

Reviewer: gotcha. now I see it. thanks!

See [error](/docs/concepts/inline-hooks/#error) for general information about the structure to use for the `error` object.

For the Registration Inline Hook, the `error` object provides a way of displaying an error message to the end user who is trying to register or update their profile.

> **Note:** Generic Progressive Enrollment error messages are only available in the Identity Engine.

* If you're using the Okta Sign-In Widget for Profile Enrollment, only the `errorSummary` messages of the `errorCauses` objects that your external service returns appear as inline errors, given the following:

   * You don't customize the error handling behavior of the widget.

>    * The `location` of `errorSummary` in the `errorCauses` object specifies the request object's user profile attribute.

Reviewer 2: I question this bullet point. Namely, even if it's required. I'd need to do some testing, but leaving this attribute blank, I believe, will not have an affect on where the UI message appears. Or, can error messages be tied, inline, to the appropriate progressive profiling field? The `location` field is added from the developer as a string field to aid in troubleshooting, according to our doc. https://developer.okta.com/docs/concepts/inline-hooks/#error

Author: @ericalong-okta We worked on this together. What are your thoughts?

Reviewer 3: @brianduffield-okta it'll affect the UI error message if the `location` attribute is left blank, the improvement was requested by [OKTA-473152](https://oktainc.atlassian.net/browse/OKTA-473152). The error messages are now tied and inline to SIW fields for both registration (v1), SSR and PE (oie).
